### PR TITLE
etcd-restore-operator: add etcd-restore-operator controller

### PR DESCRIPTION
**Pull request by fanminshi**
add controller portion of etcd-restore-operator.
this is a framework for setting up receiving/retry restore CR request.

### CODE REVIEW

**Comment by fanminshi** [comment]
PTAL cc/ @hongchaodeng @hasbro17

**Comment by hongchaodeng** [comment]
lgtm

**Comment by fanminshi** [comment]
manual test:

create cr:
```yaml
apiVersion: "etcd.database.coreos.com/v1beta2"
kind: "EtcdRestore"
metadata:
  name: example-etcd-cluster
```

etcd-backup-operator output:
```
time="2017-10-12T18:01:20Z" level=info msg="starting restore controller" pkg=controller
&{{ } {example-etcd-cluster  default /apis/etcd.database.coreos.com/v1beta2/namespaces/default/etcdrestores/example-etcd-cluster 64538c9f-af77-11e7-a83b-080027dc7a4d 305635 0 2017-10-12 18:01:37 +0000 UTC <nil> <nil> map[] map[] [] nil [] } {{0   false <nil> <nil> <nil> <nil> <nil>} {  {<nil>}}} {false }}
```

**Comment by fanminshi** [comment]
I added run controller logic in main.go
PTAL cc/ @hongchaodeng

File: pkg/etcd-restore-operator/controller/controller.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
+// Copyright 2017 The etcd-operator Authors
```
Comment (xiang90) [inline pkg/etcd-restore-operator/controller/controller.go]: we should remove etcd mentioning in all dir/file names. it is redundant.

Comment (fanminshi) [reply pkg/etcd-restore-operator/controller/controller.go]: sounds good.

**Comment by fanminshi** [comment]
all fixed PTAL cc/ @xiang90

**Comment by hasbro17** [comment]
LGTM

**Pull request merged**